### PR TITLE
Feature/roc 7041 rpa callback

### PR DESCRIPTION
**Pull request by akritis22**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-7041

### Change description ###
This is to add a resend RPA button to the CCD UI so that the case workers can do it themselves and we don't need to do it manually through the scripts.

**Does this PR introduce a breaking change?**

```
- [ ] Yes
- [x ] No
```

### CODE REVIEW

File: definition/data/sheets/CaseField.json [hunk 1]
```diff
@@ -92,5 +92,6 @@
   {"CaseTypeID": "MoneyClaimCase", "ID": "directionOrder", "Label": "Direction order", "FieldType": "DirectionOrder", "SecurityClassification": "Public"},
   {"CaseTypeID": "MoneyClaimCase", "ID": "channel", "Label": "Channel", "FieldType": "FixedList", "FieldTypeParameter": "ChannelType", "SecurityClassification": "Public"},
   {"CaseTypeID": "MoneyClaimCase", "ID": "intentionToProceedDeadline", "Label": "Intention To Proceed Deadline", "FieldType": "Date", "FieldTypeParameter": "ChannelType", "SecurityClassification": "Public"},
-  {"CaseTypeID": "MoneyClaimCase", "ID": "state", "Label": "state", "FieldType": "Text", "SecurityClassification": "Public"}
+  {"CaseTypeID": "MoneyClaimCase", "ID": "state", "Label": "state", "FieldType": "Text", "SecurityClassification": "Public"},
+  {"CaseTypeID": "MoneyClaimCase", "ID": "RPAEventType", "Label": "Reset Pin", "FieldType": "FixedList", "FieldTypeParameter": "RPAType", "SecurityClassification": "Public"}
```
Comment (daveoh) [inline definition/data/sheets/CaseField.json]: The label should be `Resend RPA`

File: definition/data/sheets/CaseEventToFields.json [hunk 2]
```diff
@@ -74,6 +74,6 @@
...
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "CaseFieldID": "respondents", "PageID": 1, "PageDisplayOrder": 1, "DisplayContext": "COMPLEX", "PageFieldDisplayOrder": 1},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "CaseFieldID": "staffUploadedDocuments", "PageID": 2, "PageDisplayOrder": 2,"DisplayContext": "COMPLEX", "PageLabel":"Upload mediation agreement","PageFieldDisplayOrder": 1},
-  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "FailedMediation", "CaseFieldID": "respondents", "PageID": 1, "PageDisplayOrder": 1, "DisplayContext": "COMPLEX", "PageFieldDisplayOrder": 1}
-
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "FailedMediation", "CaseFieldID": "respondents", "PageID": 1, "PageDisplayOrder": 1, "DisplayContext": "COMPLEX", "PageFieldDisplayOrder": 1},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "ResetRpa", "CaseFieldID": "RPAEventType", "PageID": 1, "PageDisplayOrder": 1, "DisplayContext": "MANDATORY", "PageFieldDisplayOrder": 1}
```
Comment (daveoh) [inline definition/data/sheets/CaseEventToFields.json]: CaseEventID should be ResendRpa

File: definition/data/sheets/CaseEvent.json [hunk 3]
```diff
@@ -65,5 +65,6 @@
...
+  {"CaseTypeID": "MoneyClaimCase", "ID": "CreateCitizenClaim", "Name": "Claim created by citizen", "Description": "Claim created by citizen", "DisplayOrder": 1,"PreConditionState(s)": "awaitingCitizenPayment","PostConditionState": "create", "SecurityClassification": "Public", "CallBackURLAboutToSubmitEvent": "${CCD_DEF_CLAIM_STORE_BASE_URL}/cases/callbacks/about-to-submit", "RetriesTimeoutAboutToSubmitEvent": 0,"CallBackURLSubmittedEvent": "${CCD_DEF_CLAIM_STORE_BASE_URL}/cases/callbacks/submitted", "RetriesTimeoutURLSubmittedEvent": 0},
+  {"CaseTypeID": "MoneyClaimCase", "ID": "ResetRpa", "Name": "Reset RPA", "Description": "Reset and send pin to defendant", "DisplayOrder": 1,"PreConditionState(s)": "*","PostConditionState": "*", "SecurityClassification": "Public", "CallBackURLAboutToSubmitEvent": "${CCD_DEF_CLAIM_STORE_BASE_URL}/cases/callbacks/about-to-submit", "RetriesTimeoutAboutToSubmitEvent": 0}
```
Comment (daveoh) [inline definition/data/sheets/CaseEvent.json]: ID should be ResendRpa

File: definition/data/sheets/AuthorisationCaseEvent.json [hunk 4]
```diff
@@ -252,5 +252,7 @@
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "ResumeClaimPaymentCitizen", "UserRole": "citizen", "CRUD": "CR"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "ResumeClaimPaymentCitizen", "UserRole": "caseworker-cmc", "CRUD": "C"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "CreateCitizenClaim", "UserRole": "citizen", "CRUD": "CR"},
-  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "CreateCitizenClaim", "UserRole": "caseworker-cmc", "CRUD": "CR"}
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "CreateCitizenClaim", "UserRole": "caseworker-cmc", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "ResetRpa", "UserRole": "caseworker-cmc", "CRUD": "CR"}
```
Comment (daveoh) [inline definition/data/sheets/AuthorisationCaseEvent.json]: CaseEventID should be ResendRpa

File: definition/data/sheets/CaseField.json [hunk 5]
```diff
@@ -92,5 +92,6 @@
   {"CaseTypeID": "MoneyClaimCase", "ID": "directionOrder", "Label": "Direction order", "FieldType": "DirectionOrder", "SecurityClassification": "Public"},
   {"CaseTypeID": "MoneyClaimCase", "ID": "channel", "Label": "Channel", "FieldType": "FixedList", "FieldTypeParameter": "ChannelType", "SecurityClassification": "Public"},
   {"CaseTypeID": "MoneyClaimCase", "ID": "intentionToProceedDeadline", "Label": "Intention To Proceed Deadline", "FieldType": "Date", "FieldTypeParameter": "ChannelType", "SecurityClassification": "Public"},
-  {"CaseTypeID": "MoneyClaimCase", "ID": "state", "Label": "state", "FieldType": "Text", "SecurityClassification": "Public"}
+  {"CaseTypeID": "MoneyClaimCase", "ID": "state", "Label": "state", "FieldType": "Text", "SecurityClassification": "Public"},
+  {"CaseTypeID": "MoneyClaimCase", "ID": "RPAEventType", "Label": "Resend RPA", "FieldType": "FixedList", "FieldTypeParameter": "RpaType", "SecurityClassification": "Public"}
```
Comment (karthikkannankainos) [inline definition/data/sheets/CaseField.json]: This is not required in my opinion to store RPAType that is retriggered. If we need it, there are many times we an retrigger the RPA, so we have to log all the time we have sent RPA.

Comment (dharmendrak) [reply definition/data/sheets/CaseField.json]: if we are populating this field in about to start and showing in page, then this field should be here.

**Pull request merged**